game/src/bw_1161.rs

```diff
@@ -0,0 +1,612 @@
+            error!(
+                "Couldn't read replay folder '{}': {}",
+                replay_folder.display(),
+                e
+            );
+            return INVALID_HANDLE_VALUE;
+        }
+    };
+    let mut count = 0;
+    for entry in entries {
+        let entry = match entry {
+            Ok(o) => o,
+            Err(e) => {
+                error!(
+                    "Couldn't read replay folder '{}': {}",
+                    replay_folder.display(),
+                    e
+                );
+                return INVALID_HANDLE_VALUE;
+            }
+        };
+        let path = entry.path();
+        if path.extension() == Some(OsStr::new("rep")) {
+            if let Some(file_stem) = path.file_stem() {
+                count = count.max(initial_number(file_stem));
+            }
+        }
+    }
+    let filename = format!(
+        "{:04}_{}.rep",
+        count + 1,
+        chrono::Local::now().format("%Y-%m-%d")
+    );
+    CreateFileW(
+        windows::winapi_str(replay_folder.join(filename)).as_ptr(),
+        access,
+        share_mode,
+        security_attributes as *mut _,
+        creation_disposition,
+        flags,
+        template,
+    )
+}
+
+fn initial_number(path: &OsStr) -> u32 {
+    use std::os::windows::ffi::OsStrExt;
+    // Trickery to parse the initial number without assuming UTF-8 filename..
+    path.encode_wide()
+        .take_while(|&x| x >= b'0' as u16 && x <= b'9' as u16)
+        .fold(0, |old, new| {
+            old.wrapping_mul(10)
+                .wrapping_add((new - b'0' as u16) as u32)
+        })
+}
+
+#[test]
+fn test_initial_number() {
+    assert_eq!(initial_number(OsStr::new("123asdhk")), 123);
+    assert_eq!(initial_number(OsStr::new("asd")), 0);
+    assert_eq!(initial_number(OsStr::new("a234sd")), 0);
+    assert_eq!(initial_number(OsStr::new("1")), 1);
+    assert_eq!(initial_number(OsStr::new("001241.rep")), 1241);
+    assert_ne!(
+        initial_number(OsStr::new("100000000000000000001241.rep")),
+        0
+    );
+}
+
+fn get_documents_path() -> Result<PathBuf, io::Error> {
+    use winapi::um::combaseapi::CoTaskMemFree;
+    use winapi::um::knownfolders::FOLDERID_Documents;
+    use winapi::um::shlobj::SHGetKnownFolderPath;
+
+    unsafe {
+        let mut path = null_mut();
+        let error = SHGetKnownFolderPath(&FOLDERID_Documents, 0, null_mut(), &mut path);
+        if error != 0 {
+            return Err(io::Error::from_raw_os_error(error));
+        }
+        let len = (0..).find(|&x| *path.add(x) == 0).unwrap();
+        let slice = std::slice::from_raw_parts(path, len);
+        let result = windows::os_string_from_winapi(slice);
+        CoTaskMemFree(path as *mut _);
+        Ok(result.into())
+    }
+}
+
+unsafe fn create_lobby(
+    map_path: &Path,
+    name: &str,
+    game_type: bw::GameType,
+) -> Result<(), bw::LobbyCreateError> {
+    let map = find_map_entry(map_path)?;
+    // Password must be null for replays to work
```

Comment:
Is this correct? I thought when watching replays through a lobby, password must *not* be null?
Or is this related to something else?

Reply:
You cannot set password anyway in LAN, but for some reason hosting replay will specifically check that there isn't password.. I think? Anyway it's been like that always here